src/io_trait.rs

```diff
@@ -0,0 +1,35 @@
+use crate::error::Error;
+
+pub trait JkReader: JkReaderClone {
+    fn read_to_string(&self, path: &str) -> Result<String, Error>;
+}
+
+pub trait JkReaderClone {
+    fn box_clone(&self) -> Box<dyn JkReader>;
+}
+
+impl<T> JkReaderClone for T
+where
+    T: 'static + JkReader + Clone,
+{
+    fn box_clone(&self) -> Box<dyn JkReader> {
+        Box::new(self.clone())
+    }
+}
+
+impl Clone for Box<dyn JkReader> {
+    fn clone(&self) -> Self {
+        self.box_clone()
+    }
+}
+
+#[cfg(feature = "std")]
+#[derive(Clone)]
+pub struct JkStdReader {}
```

Comment:
```suggestion
pub struct JkStdReader;
```
This will also make it easier and cleaner to instantiate when creating `Context`s

Reply:
Fixed by 282dce49e18f7cfb41903a2abce277c4d0df57df